File: Expt12_Auth_App/lib/screens/home_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../services/auth_service.dart';

class HomeScreen extends StatelessWidget {
  const HomeScreen({super.key});
  @override
  Widget build(BuildContext context) {
    final auth = Provider.of<AuthService>(context, listen: false);
    final user = auth.currentUser;
    return Scaffold(
      appBar: AppBar(
        title: const Text('Home'),
        actions: [
          IconButton(
            icon: const Icon(Icons.logout),
            onPressed: () async {
              await auth.signOut();
            },
          ),
        ],
      ),
      body: Center(child: Text('Welcome, ${user?.email ?? 'User'}')),
    );
  }
}

File: Expt12_Auth_App/React_Auth_App/screens/HomeScreen.js
import { signOut } from 'firebase/auth';
import React from 'react';
import { Button, Text, View } from 'react-native';
import { auth } from '../config/firebaseconfig';

export default function HomeScreen() {
  const user = auth.currentUser;

  return (
    <View style={{ padding: 20 }}>
      <Text style={{ fontSize: 18, marginBottom: 12 }}>Welcome, {user?.email ?? user?.phoneNumber ?? 'User'}</Text>
      <Button title="Logout" onPress={() => signOut(auth)} />
    </View>
  );
}

